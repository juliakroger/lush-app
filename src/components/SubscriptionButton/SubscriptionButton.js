import React from 'react'
import styles from './SubscriptionButton.module.scss'

const SubscriptionButton = () => {
  return (
    <div className={styles.newsletterButton}>
      <input
        id='newsletter-email'
        type='email'
        name='email'
        placeholder='Your email'
        className={styles.subscribeInput}
      ></input>

      <button id='subscribe-button' className={styles.subscribeButton}>
        Subscribe
      </button>
    </div>
  )
}

export default SubscriptionButton
