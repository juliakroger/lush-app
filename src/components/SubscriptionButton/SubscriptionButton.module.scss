@import '../../scss/variables.scss';
@import '../../scss/responsive-utils';

.newsletterButton {
  flex-direction: row;
  display: flex;
  width: 100%;
  padding-top: 20px;

  .subscribeInput {
      font-size: 12px;
    width: 300px;
    border: 1px solid #e9eaec;
    border-radius: 5px;
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
    padding: 13px 16px 11px 16px !important;
  }

  .subscribeButton {
    font: inherit;

    cursor: pointer;
    border-radius: 5px;
    background-color: $super-orange-color;
    border: 1px solid transparent;
    color: #fff;
    font-size: 14px;

    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
  }
}
