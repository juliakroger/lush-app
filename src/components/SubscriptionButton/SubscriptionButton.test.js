import React from 'react'
import { shallow } from 'enzyme'
import SubscriptionButton from './SubscriptionButton'

describe('Component - Subscription Button', () => {
  it('should render without crashing', () => {
    const component = shallow(<SubscriptionButton />)
    expect(component).toMatchSnapshot()
  })

  it('should trigger button correctly', () => {
    const mockFunc = jest.fn()
    const component = shallow(<SubscriptionButton clickSubscribe={mockFunc} />)

    component.find('button').simulate('click')
    expect(mockFunc.mock.calls.length).toBe(1)
  })
})
