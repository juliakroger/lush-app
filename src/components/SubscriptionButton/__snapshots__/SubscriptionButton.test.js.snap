// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Subscription Button should render without crashing 1`] = `
<div
  className="newsletterButton"
>
  <input
    className="subscribeInput"
    id="newsletter-email"
    name="email"
    placeholder="Your email"
    type="email"
  />
  <button
    className="subscribeButton"
    id="subscribe-button"
    onClick={[Function]}
  >
    Subscribe
  </button>
</div>
`;
